src/sa/sa.c

```diff
@@ -591,3 +591,42 @@ bool sa_is_any(const struct sa *sa)
 		return false;
 	}
 }
+
+
+#ifdef HAVE_INET6
+void sa_set_scopeid(struct sa *sa, int scopeid)
+{
+	if (!sa)
+		return;
+
+	sa->u.in6.sin6_scope_id = scopeid;
+}
+
+
+int sa_scopeid(const struct sa *sa)
+{
+	if (!sa)
+		return 0;
+
+	return sa->u.in6.sin6_scope_id;
+}
+
+
+int sa_cpy_scopeid(struct sa *sa_dst, const struct sa *sa_src)
+{
+	if (!sa_dst || !sa_src)
+		return EINVAL;
+
+	if (!sa_is_ipv6ll(sa_dst) || !sa_is_ipv6ll(sa_src))
+		return EINVAL;
+
+	sa_dst->u.in6.sin6_scope_id = sa_src->u.in6.sin6_scope_id;
+	return 0;
+}
+
+
+bool sa_is_ipv6ll(const struct sa *sa)
```

Comment:
not sure if this function is needed.

the check can be done in the application

Reply:
Since it is used several times it reduces the code. Your next question could make it obsolete.